### PR TITLE
Created base structure

### PR DESCRIPTION
**Pull request by bones-wp**

### CODE REVIEW

File: src/main/java/com/aston/internship/entity/Feedback.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    private Long id;
+
+    private Double stars;
+
+    @Column(length = 10000)
+    private String message;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    private Producer producer;
+
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Feedback.java:31]: Add Customer customer here

File: src/main/java/com/aston/internship/entity/Lashes.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@Table(name = "LASHES")
+@AllArgsConstructor
+@Data
+
+public class Lashes {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    String type;
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Lashes.java]: Add access modifier for each field

File: src/main/java/com/aston/internship/entity/Producer.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Embedded
+    private Details details;
+
+    @Embedded
+    private SocialNetwork socialNetworks;
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Producer.java:34]: Maybe List of String socialNetworks?

File: src/main/java/com/aston/internship/entity/Notation.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+public class Notation {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private Calendar calendar;
+
+    public Notation() {
+
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Notation.java:34]: Add producerId
@ManyToOne(fetch = FetchType.LAZY)
@JoinColumn(name = "producer_id")
Long producerId;
The same for consumer
Also see comments for it's class for corresponding fields

File: src/main/java/com/aston/internship/entity/Customer.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+public class Customer {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Embedded
+    private Details details;
+
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinColumn(name = "customer_id")
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Customer.java:31]: Use  @OneToMany(... , mappedBy = "customerId")

File: src/main/java/com/aston/internship/entity/Producer.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinColumn(name = "master_id")
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Producer.java]: Use  @OneToMany(... , mappedBy = "producerId")

File: src/main/java/com/aston/internship/entity/LashesPrice.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne
+    private Lashes lashes;
+
+    private Double price;
+
+    public LashesPrice() {
+
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/LashesPrice.java:29]: Add Long producerId

File: src/main/java/com/aston/internship/entity/Producer.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    private List<LashesPrice> lashesPrice;
+
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinColumn(name = "master_id")
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Producer.java]: Use @OneToMany(... , mappedBy = "producerId")

File: src/main/java/com/aston/internship/entity/Customer.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    private List<Notation> schedule;
+
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Customer.java:33]: Add List of Feedback feedbacks

File: src/main/java/com/aston/internship/entity/Producer.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    private List<Notation> schedule;
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Producer.java:42]: Rename with notations

File: src/main/java/com/aston/internship/entity/Producer.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Embedded
+    private Details details;
+
+    @Embedded
+    private SocialNetwork socialNetworks;
+
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinColumn(name = "master_id")
+    private List<LashesPrice> lashesPrice;
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Producer.java]: Use plural form of "lashesPrice"

File: src/main/java/com/aston/internship/entity/Customer.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Embedded
+    private Details details;
+
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinColumn(name = "customer_id")
+    private List<Notation> schedule;
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Customer.java]: Rename with notations

File: src/main/java/com/aston/internship/entity/Producer.java [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinColumn(name = "master_id")
+    private List<Notation> schedule;
+
+    private Double avRate;
+
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Producer.java:49]: Add List of Feedback feedbacks

File: src/main/java/com/aston/internship/entity/Feedback.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private Double stars;
+
+    @Column(length = 10000)
+    private String message;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    private Producer producer;
```
Comment (intensiveproject) [inline src/main/java/com/aston/internship/entity/Feedback.java:30]: @JoinColumn(name = "producer_id")

**Review by intensiveproject — CHANGES_REQUESTED** [review]

File: src/main/java/com/aston/internship/entity/Feedback.java (hunk 1 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Feedback.java:31]: Ok

File: src/main/java/com/aston/internship/entity/Lashes.java (hunk 2 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Lashes.java]: 👌🏻 🙈

File: src/main/java/com/aston/internship/entity/Lashes.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    String form;
+    Double length;
```
Comment (MikhailBusygin) [inline src/main/java/com/aston/internship/entity/Lashes.java]: length = width

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 3 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Producer.java:34]: It doesn't work like that

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 13 above)
Comment (MikhailBusygin) [reply src/main/java/com/aston/internship/entity/Producer.java:49]: @Column(name = "av_rate") над avRate

File: src/main/java/com/aston/internship/entity/Notation.java (hunk 4 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Notation.java:34]: Эти связи уже есть в сущностях Producer и Customer. См. схему БД

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 10 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Producer.java:42]: 👌🏻

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 13 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Producer.java:49]: Ok

File: src/main/java/com/aston/internship/entity/Customer.java (hunk 12 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Customer.java]: Что переименовать?

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 11 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Producer.java]: Ок

File: src/main/java/com/aston/internship/entity/Customer.java (hunk 5 above)
Comment (MikhailBusygin) [reply src/main/java/com/aston/internship/entity/Customer.java:31]: mappedBy = "customer_id"

File: src/main/java/com/aston/internship/entity/Feedback.java (hunk 14 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Feedback.java:30]: Сделал в Producer

File: src/main/java/com/aston/internship/entity/Lashes.java (hunk 15 above)
Comment (bones-wp) [reply src/main/java/com/aston/internship/entity/Lashes.java]: У них вроде нет градации по ширине, по длине только видел

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 6 above)
Comment (MikhailBusygin) [reply src/main/java/com/aston/internship/entity/Producer.java]: mappedBy = "producer_id"

File: src/main/java/com/aston/internship/entity/Producer.java (hunk 8 above)
Comment (MikhailBusygin) [reply src/main/java/com/aston/internship/entity/Producer.java]: mappedBy = "producer_id"

**Pull request merged**